### ISSUE TITLE
Device statuses are Unavailable

### ISSUE THREAD

**Issue by LyZzardMD** [peterbuga/HASS-sonoff-ewelink#184]
After the latest updates this custom component is not stable. I've noticed that after a restart of home assistant the devices are available for about 3 hours, after that all of them become Unavailable. Did not find a solution to this problem.

**Comment by sanjeewasam** [comment]
I am still with 0.98. However today I noted that status update is not avilable. if I add region code say us then sensors are populated but switches not there yet. If I add all optional fiields still same. This used to work well. Not sure why this is breaking now

**Comment by sanjeewasam** [comment]
OK. I am in Australia and 0.98.2, using the following it is working for me as before 
sonoff:
  username: myemail@gmail.com
  password: mypassword
  scan_interval: 90 #(optional, lower values than 60 won't work anymore!)
  grace_period: 600 #(optional)
  api_region: 'as' #(optional)
  entity_prefix: True #(optional)
  debug: False #(optional)

**Comment by LyZzardMD** [comment]
![Screenshot_20191128-230602](https://user-images.githubusercontent.com/3432427/69831784-b060fe00-1233-11ea-95f8-bf1a1914eb64.png)
![Screenshot_20191128-230539](https://user-images.githubusercontent.com/3432427/69831793-b8b93900-1233-11ea-981b-c17a05537b98.png)

I think that at restart the session is being started and it expires in several hours. It is not reconnected. These are 2 sonoff devices.

**Comment by peterbuga** [comment]
i have no problem `¯\_(ツ)_/¯`

@LyZzardMD do you use a 2nd eWeLink account?

**Comment by LyZzardMD** [comment]
No. The original one. I tried to share items but it did not work. I signed out from the app to leave active session only on HA and it is the same.

**Comment by peterbuga** [comment]
@LyZzardMD try to do more tests and see if it happens all the time the same, i.e. going unavailable after X period. it's important to figure out if it's just a _you problem_ or not (TBH i would expect more people complaining about it if so)

**Comment by DPAmes** [comment]
I think I have the same problem. I've been using this component successfully for about 6 months with a single Sonoff-type switch supported by eWeLink (thanks Peter!). Everything was fine until yesterday, when the switch became Unavailable due to the sonoff component not initializing successfully. I've tested everything I can think of. I tried downloading the Oct fix, but it just made things worse - HA won't start at all (maybe because of older version 0.92?). This is the error reported by HA (with the original component files):

```Error during setup of component sonoff

Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.6/site-packages/homeassistant/setup.py", line 153, in _async_setup_component
    hass, processed_config)
  File "/home/homeassistant/.homeassistant/custom_components/sonoff/__init__.py", line 53, in async_setup
    hass.data[DOMAIN] = Sonoff(hass, config)
  File "/home/homeassistant/.homeassistant/custom_components/sonoff/__init__.py", line 108, in __init__
    self.do_login()
  File "/home/homeassistant/.homeassistant/custom_components/sonoff/__init__.py", line 192, in do_login
    self.update_devices() # to write the devices list
  File "/home/homeassistant/.homeassistant/custom_components/sonoff/__init__.py", line 357, in update_devices
    if self.is_grace_period():
  File "/home/homeassistant/.homeassistant/custom_components/sonoff/__init__.py", line 322, in is_grace_period
    grace_time_elapsed = self._skipped_login * int(SCAN_INTERVAL.total_seconds())
NameError: name 'SCAN_INTERVAL' is not defined```

**Comment by peterbuga** [comment]
with all due respect @DPAmes i'm going to ignore your comment as you mixed versions too much and i can't offer a possible fix, unless you're willing to update to latest HA & sonoff there's much i can do for you

@LyZzardMD any updates from you?

**Comment by DPAmes** [comment]
No response needed for me - in case I wasn't clear, the October update to the sonoff component fixed my problem. No issue to do with HA version. I left my message above only in case it helps someone confirm that they have the same problem and the same fix will likely work for them.

**Comment by LyZzardMD** [comment]
I still have the issue. Please note that I have 2 HA instances one in Docker and one in Raspberry Pi. I switched one ON and other OFF to see if the issue is still there and Yes, it is still there.

The issue is still present. On my fresh install in Docker I have the issue again. I noticed that everytime it becomes Unavailable at the same hour. It is 03:00 AM (EET). I thing that the login session expires or is being reset by HA or eWelink and it is not reestablished.

Unfortunately I did not understand well the code of the component to add some debug lines in it to give you more details. Find bellow the screenshot from today:

![DeepinScreenshot_select-area_20191206111129](https://user-images.githubusercontent.com/3432427/70311028-82724f80-1819-11ea-805a-198913d7941a.png)

**Comment by peterbuga** [comment]
@LyZzardMD please find a way to create a 2nd ewelink account and share the devices and try that account *exclusively only* on one HA instance and nothing more (do not login with this 2nd account on another mobile or use it another HA config)
i would not look further into this problem until you achieve this and re-confirm the problem persists

**Comment by LyZzardMD** [comment]
I guess all is fine. I understood the problem when I realised that they become Unavailable at 3 o'clock. And I remembered that my was unstable and I've created an automation to restart HA at 3:00 everyday. There are no other automations left on that Pi, but the configurations are present and entities also. I saw that the entities are also randomly becoming unavailable also on Pi and on my docker instance and I thought it is a global problem. Now I know that they were disconnecting each other. Thank you for patience :)
